modules/dns_nsupdate/dns_nsupdate_gss_main.rb

```diff
@@ -6,17 +6,12 @@ class Record < Proxy::Dns::Nsupdate::Record
     include Proxy::Kerberos
     attr_reader :tsig_keytab, :tsig_principal
 
-    def initialize options = {}
+    def initialize(server = nil, ttl = nil)
       @tsig_keytab = ::Proxy::Dns::NsupdateGSS::Plugin.settings.dns_tsig_keytab
       @tsig_principal = ::Proxy::Dns::NsupdateGSS::Plugin.settings.dns_tsig_principal
-      raise "Keytab not configured via dns_tsig_keytab for DNS GSS-TSIG support" unless tsig_keytab
-      raise "Unable to read dns_tsig_keytab file at #{tsig_keytab}" unless File.exist?(tsig_keytab)
-      raise "Kerberos principal required - check dns_tsig_principal setting" unless tsig_principal
```

Comment:
There is a default value for it here: https://github.com/theforeman/smart-proxy/blob/develop/modules/dns_nsupdate/dns_nsupdate_gss_plugin.rb#L9. My reasoning here (and in other places where I removed similar nil checks) is that once default value is defined it's pretty unlikely it will ever be nil. It possible though to make a mistake in the config file (leave option without a value), I should filter out settings pairs without values before merging them with default values (here: https://github.com/theforeman/smart-proxy/blob/develop/lib/proxy/settings/plugin.rb#L10 and here: https://github.com/theforeman/smart-proxy/blob/develop/lib/proxy/settings/global.rb#L30)